py/desisim/templates.py

```diff
@@ -1827,8 +1836,13 @@ def make_templates(self, nmodel=100, zrange=(0.5, 4.0), rmagrange=(20.0, 22.5),
         for key, value in zip(('REDSHIFT', 'MAG', 'SEED'),
                                (redshift, mag, templateseed)):
             meta[key] = value
-        if lyaforest:
+        if lyaforest: 
             meta['SUBTYPE'] = 'LYA'
+        if balqso: 
+            if lyaforest: 
+                meta['SUBTYPE'] = 'LYA+BAL'
+            else: 
+                meta['SUBTYPE'] = 'BAL'
 
```

Comment:
@paulmartini Shouldn't the SUBTYPE be set to LYA+BAL only if the spectrum contains a BAL, i.e., after `balprob` (the BAL probability) has been applied?

Reply:
I suggest we merge as soon as the template file issue is solved. This PR has been opened more than 3 month ago; the basic functionalities are in place and the Lya WG would need this to move on: this code is needed to test the BAL identifiers, the impact on redshift efficiency, correlation function ... etc.

Reply:
This code is not ready to merge yet and--again--it is not holding up any of the Lya QSO activities because this PR is not integrated with SIMQSO.  I'll merge as soon as it's ready but right now the output spectra are not reproducible even with an input seed, which needs to be resolved.

Reply:
BALs are integrated in quickquasars in the branch qqso_dla_bal which has been branched from this one. We are now discussing about adding metals. We have some sort of a branch traffic jam here.

Reply:
I opted to set SUBTYPE to 'LYA+BAL' if balqso=True for the simulated quasars, even though not all quasars will be BALs (unless balprob=1). My rationale was that someone may want to know if the balqso flag were True when a set of simulated quasars were generated. I intended to eventually create a separate index (to be stored in META) that would either have the template ID (if a BAL) or be empty, but I have not figured out how to do that yet.

Reply:
@moustakas With regard to @julienguy 's comment, there is another group (not me) in the Ly-a WG who will add BALs to SIMQSO. I think their work will be easier if the issues with this implementation are resolved first.

Reply:
Julien, I can comment out the pieces for the BALs and move to another branch that does not depend on this one until this is completed... I branched from the BAL's branch only because I was adding the BALS, but the other pieces don't need this...Also, maybe for quickquasars we don't need to add the BALs at the level of SIMQSO, could it be added to the transmission as we did with the DLAs?